src/app/features/nonce/use-persisted-nonce.tsx

```diff
@@ -0,0 +1,41 @@
+import { useState } from 'react';
+
+import { isDefined, isUndefined } from '@shared/utils';
+
+import { useOnMount } from '@app/common/hooks/use-on-mount';
+import { useNextNonce } from '@app/query/stacks/nonce/account-nonces.hooks';
+
+export function usePersistedNonce() {
+  const [persistedNonce, setPersistedNonce] = useState(0);
+  const { data: nextNonce } = useNextNonce();
+
+  useOnMount(async () => {
+    const state = await chrome.storage.session.get('nonce');
+    const persistedNonceState = state['nonce'];
+
+    if (!persistedNonceState && isDefined(nextNonce)) {
+      setPersistedNonce(nextNonce.nonce ?? 0);
+      await chrome.storage.session.set({ nonce: nextNonce.nonce });
```

Comment:
Recommend saving nonce values in a sturcture like:

```
{ [address]: number }
```

otherwise, we'll save the wrong nonce value when changing accounts?

Reply:
I would extra this into some helper methods. To maintain state from other accounts, we might have to read state first, then set the new state spread operating the existing